ostree-compliance-mode.spec

```diff
@@ -26,14 +27,19 @@ ${CC} ${RPM_OPT_FLAGS} -Wall -Winvalid-pch -Wextra -Wpedantic -Werror -std=gnu11
 %install
 install -Dm 4411 ostree-compliance-mode %{buildroot}%{_bindir}/ostree-compliance-mode
 install -Dm 500 ostree-compliance-mode-helper %{buildroot}%{_libexecdir}/ostree-compliance-mode-helper
+install -Dm 600 <("") %{buildroot}%{_sysconfdir}/ostree-compliance-mode.conf
 
```

Comment:
The idea is correct, would it be possible to do this here (copying your change from above):

```
%post
rpm-ostree status --json | python -c "import json,sys; print(json.load(sys.stdin)[\"deployments\"][$i][\"origin\"])" > %{buildroot}%{_sysconfdir}/ostree-compliance-mode.conf
```

This way we can guarantee it's only done once and in the built ostree image layer without the protected content. Rather than whenever we detect the file is not present or empty. We don't require the `if test -s` this way either.

The issue is if we populate this on first execution after install, we populate with the ref of the version with the protected content, we don't want to give modifiable access to that version.

Reply:
You'd have to do the loopy thing to populate the i variable also of course, I wouldn't do an `exit 4` in this case though, just printing "No booted deployment found\n" and leaving the file unpopulated should be enough

Reply:
So it looks like this solution will not work. I've also removed the changes to the ostree-compliance-mode-helper as you suggested. Thanks for the explanation above.
As discussed, the ostree-compliance-mode.conf file may be populated outside of the package.